src/DrupalDoctrineCache.php

```diff
@@ -26,82 +54,63 @@ public function fetch($id) {
   }
 
   /**
-   * Checks if the Drupal cache contains the specified entry.
-   *
-   * @inheritdoc
+   * {@inheritdoc}
    */
-  public function contains($id) {
-    $entry = $this->fetch($id);
+  protected function doContains($id) {
+    $entry = $this->doFetch($id);
     return !empty($entry);
   }
 
   /**
-   * Saves data to the Drupal cache.
-   *
-   * @inheritdoc
+   * {@inheritdoc}
    */
-  public function save($id, $data, $lifeTime = NULL) {
+  protected function doSave($id, $data, $lifeTime = 0) {
     // Doctrine defines NULL to mean permanent cache.
-    if ($lifeTime == NULL) {
+    if ($lifeTime === NULL) {
       $lifeTime = CACHE_PERMANENT;
     }
-    else {
+    elseif ($lifeTime > 0) {
       $lifeTime += time();
     }
 
     // Cache data, converting Doctrine lifetime to a unix timestamp for Drupal.
-    $result = cache_set($this->getCacheId($id), $data, $this->cache_table, $lifeTime);
+    $result = $this->cache_set($this->getCacheId($id), $data, $this->cache_table, $lifeTime);
     return empty($result) ? FALSE : TRUE;
   }
 
   /**
-   * Deletes the entry from the Drupal cache.
+   * Deletes a cache entry.
+   *
+   * @param string $id The cache id.
    *
-   * @inheritdoc
+   * @return boolean TRUE if the cache entry was successfully deleted, FALSE otherwise.
    */
-  public function delete($id) {
-    cache_clear_all($this->getCacheId($id), $this->cache_table);
+  protected function doDelete($id) {
+    $this->cache_clear_all($this->getCacheId($id), $this->cache_table);
     return TRUE;
   }
 
   /**
-   * Not implemented in this cache.
-   *
-   * Original description :-
+   * Flushes all cache entries.
    *
-   * @inheritdoc
+   * With Drupal, caches are not really cleared until cron is run.
    *
+   * @return boolean TRUE if the cache entries were successfully flushed, FALSE otherwise.
    */
-  public function getStats() {
-    // We don't implement this.
-    return NULL;
-  }
-
-  /**
-   * Generate a cache ID for the cache entry, based on the original ID.
-   *
-   * @param $id
-   */
-  protected function getCacheId($id) {
-    return self::CACHE_PREFIX . "{$id}";
+  protected function doFlush() {
+    $this->cache_clear_all('*', $this->cache_table, TRUE);
+    return TRUE;
   }
 
   /**
-   * Set the cache table to be used by the caching API.
+   * Not implemented in this cache.
    *
-   * @param string $cache_table
-   */
-  public function setCacheTable($cache_table) {
-    $this->cache_table = $cache_table;
-  }
-
-  /**
-   * Get the cache table used by the caching API.
+   * Original description :-
    *
-   * @return string
+   * {@inheritdoc}
+   * @codeCoverageIgnore
```

Comment:
We don't implement doGetStats(), so it didn't make sense to test it to me. The function could also remove the return to get the same effect?